src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2189,7 +2189,7 @@ memory::ptr primitive_inst::allocate_output(engine& _engine,
             GPU_DEBUG_LOG << "[" << _node.id() << ": constant]" << std::endl;
             return ov::intel_gpu::allocate_memory_evenif_zero_bytes(_engine, layout, alloc_type, reset);
         }
-    } else if (!_node.can_share_buffer() || _node.can_be_optimized() || _node.is_output()) {
```

Comment:
@yeonbok would you please review if this condition is correct? Unluckily it sounds some _node.can_be_optimized cannot allocate from pool. I see some model with accuracy problem. Probably the dependencies is not set properly somewhere.

Reply:
Why not replacing it to 
(!_node.can_share_buffer() || can_be_optimized() || _node.is_output()) 
but removing _node.can_be_optimized() ?

Reply:
I changed it to `} else if (!_node.can_share_buffer() || impl_params.can_be_optimized() || _node.is_output()) {` as it's called from a static member function.
However the accuracy issue is still there. Details check my email.

Reply:
@yeonbok I think we can remove either `impl_params.can_be_optimized()` or `_node.can_be_optimized()` here. 
If `impl_params.can_be_optimized()==true`, it won't ask a memory allocation.